stock_change_qty_reason/wizard/stock_change_product_qty.py

```diff
@@ -0,0 +1,23 @@
+# Copyright 2016-2017 ACSONE SA/NV (<http://acsone.eu>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+"""stock_change_product_qty"""
+
+
+from odoo import models, fields, api
+
+
+class StockChangeProductQty(models.TransientModel):
+    """Class to inherit model stock.change.product.qty"""
+    _inherit = 'stock.change.product.qty'
+
+    reason = fields.Char('Reason',
+                         help='Type in a reason for the '
+                         'product quantity change')
+
+    @api.multi
+    def change_product_qty(self):
+        """Function to super change_product_qty"""
+        if self.reason:
+            this = self.with_context(change_quantity_reason=self.reason)
```

Comment:
sending the info using context is not a good idea, as I discussed. It is better to extend the stock.inventory.line with a field 'reason' and pass the value from the wizard using https://github.com/odoo/odoo/blob/11.0/addons/stock/wizard/stock_change_product_qty.py#L50

Reply:
I agree despite if I did that in the original module. Bad implementation. :smiley: